src/librustc/traits/error_reporting.rs

```diff
@@ -2214,6 +2214,10 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         }
 
         let span = self.tcx.def_span(generator_did);
+        // Do not ICE on closure typeck (#66868).
+        if let None = self.tcx.hir().as_local_hir_id(generator_did) {
```

Comment:
I wonder if this would be fixed by [these changes](https://github.com/rust-lang/rust/pull/65345/files#diff-0e12890ad498e250783f40f88a8b8ec6R2088). 

Super minor nit:

```suggestion
        if self.tcx.hir().as_local_hir_id(generator_did).is_none() {
```

Reply:
> I wonder if this would be fixed by these changes.

Seems like it could.